client/app/components/attachement_preview.coffee

```diff
@@ -0,0 +1,56 @@
+{li, img, a, i} = React.DOM
+MessageUtils = require '../utils/message_utils'
+
+
+module.exports = React.createClass
+    displayName: 'AttachmentPreview'
+
+    icons:
+        'archive'      : 'fa-file-archive-o'
+        'audio'        : 'fa-file-audio-o'
+        'code'         : 'fa-file-code-o'
+        'image'        : 'fa-file-image-o'
+        'pdf'          : 'fa-file-pdf-o'
+        'word'         : 'fa-file-word-o'
+        'presentation' : 'fa-file-powerpoint-o'
+        'spreadsheet'  : 'fa-file-excel-o'
+        'text'         : 'fa-file-text-o'
+        'video'        : 'fa-file-video-o'
+        'word'         : 'fa-file-word-o'
+
+
+    render: ->
+        if @props.previewLink
+             li key: @props.key,
+                @renderIcon()
+                a
+                    target: '_blank'
+                    href: @props.file.url,
+                        # TODO: generate a thumb instead of loading raw file
+                        img width: 90, src: @props.file.url if @props.preview
+                        @props.file.generatedFileName
```

Comment:
Hummm… unfortunately, it seems that the DS generate thumb on for _File_ documents, and attachment aren't stored as _File_. It can't get a valid `thumb` key which contains the thumb at this time. I'll suggest we deal with this improvement later :).